### PR TITLE
Add milestones UI and revisions

### PR DESCRIPTION
**Pull request by geovgy**
Built on top of PR #60 

Revisions:
- I modified the addMilestones() function in SmartInvoice.sol to reduce the number of for loops for calculating newTotal and added a require statement to ensure milestones are not added after terminationTime.
- Added unit test for terminationTime require statement
- Edited the names of tests to match theme of the previous tests

UI:
- Created a component for AddMilestones used in the modal on view invoice page. Following same rules as other modal components on that page.
- Added button to open modal with AddMilestones component
- New function to interact with smart contract's addMilestones()

To Do: deploying the new SmartInvoice contract and testing if UI can properly interact with addMilestones() function in contract

### CODE REVIEW

File: packages/contracts/contracts/SmartInvoice.sol [hunk 1]
```diff
@@ -130,6 +136,35 @@ contract SmartInvoice is
...
+
+        for (uint256 i = 0; i < amounts.length; i++) {
+            baseArray[i] = amounts[i];
+        }
+        for (uint256 i = amounts.length; i < newLength; i++) {
+            baseArray[i] = _milestones[i - amounts.length];
+            newTotal += _milestones[i - amounts.length];
+        }
+
+        total = newTotal;
```
Comment (psparacino) [inline packages/contracts/contracts/SmartInvoice.sol:158]: I didn't check this with tests yet, but on first glance I believe this is only replacing the total with the total of *only* the new Milestones and doesn't take into account the existing pre-addMilestones() total.  so on 148 initialize to 0 and on 158 total += newTotal.  Can double check later on.

Comment (geovgy) [reply packages/contracts/contracts/SmartInvoice.sol:158]: newTotal is initialized with the original total value and then adding the new milestone amounts... so either way should result in the same number. It's just a matter of which one is the most gas efficient. It passed testing but I can check to see if your suggestion is even more efficient.

Comment (geovgy) [reply packages/contracts/contracts/SmartInvoice.sol:158]: Looks like the current method of initializing newTotal = total saves an average of ~100 wei in comparison

Comment (plor) [reply packages/contracts/contracts/SmartInvoice.sol:158]: We can do another pass to optimize if we need to, but if tests are passing I would call it good at this point.

Comment (plor) [reply packages/contracts/contracts/SmartInvoice.sol:158]: Meaning, go ahead and merge and mark complete the related tasks.

Comment (psparacino) [reply packages/contracts/contracts/SmartInvoice.sol:158]: Ah right; I lapsed that it's initialized with the existing array.

**Pull request merged**